File: branches/TOP_API/php/demo/onsaleget.php
<?php
require('config.inc');

$top = new Net_Top(TOP_APPKEY, TOP_SECRET_KEY);
$req = Net_Top_Request_Item::onsaleGet(
    array(
        'fields' => array(':all'),
        'session' => 'wenbinye',
        )
    );
$res = $top->request($req);
if ( $res->isError() ) {
    echo "Something is wrong: ", $res->getMessage();
    echo $res->content();
    print_r($res->result());
}
else {
    print_r($res->result());
}

File: branches/TOP_API/perl/script/onsaleget.pl
#!/usr/bin/perl -w
# onsaleget.pl --- 

use warnings;
use strict;

use FindBin qw/$Bin/;
use lib "$Bin/../lib";

require TopConfig;
use Net::Top::Request::Item;
use Data::Dumper qw(Dumper);

Log::Log4perl->easy_init();

my $top = TopConfig->getClient();

my $session = 'aRzmY4FFAv29AGTnrP';
my $req = Net::Top::Request::Item->onsaleGet(
        fields => [':all'],
        session => $session
    );
my $res = $top->request($req);
if ( $res->is_success() ) {
    print Dumper($res->result()), "\n";
} else {
    print 'Something is wrong: ', $res->message(), "\n";
    print $res->content, "\n";
}

